asterics-web-vue/src/components/general/NavigationExpansion.vue

```diff
@@ -61,7 +61,12 @@ export default {
       return v.hasOwnProperty("dropdown");
     },
     expansionPanelState(v) {
-      this.expansion = v;
+      // this.expansion = v;
+      if (v) this.expansion = 0;
```

Comment:
why is the expansion state not boolean or simply 0/1?
why do you have to set it to null?

Reply:
Because `expansion` is "modeled" by the `<v-expansion-panel/>`, above.
The expansion panel controls this variable and sets it either to `0` or `null`.

```vue
<v-expansion-panel v-model="expansion" dark>
  <v-expansion-panel-content class="primary" @input="expansionPanelState($event)">
```

As a matter of fact, I was surprised too. :neutral_face: 

Long story short: [`<v-expansion-panel/>`](https://vuetifyjs.com/en/components/expansion-panels) set's it to either `0` or `null`.

Setting the variable `expansion` to `0` and `1` does not correspond with the semantics in this case.
In this case, `0` means that the `v-expansion-panel` is opened and `null` that it is collapsed.

---

Variables bond by the [`v-model`](https://vuejs.org/v2/api/#v-model) directive are used in vue to create a so-called [two-way data binding](https://vuejs.org/v2/guide/forms.html).


In our case, the variable `expansion` is updated by the `v-expansion-model` and by `expansionPanelState(v)`. When the navigation is changed by the user, it is updated by the respective component (i.e. `<v-expansion-panel/>`) when an "internal value" changes.

At the same time, when the variable `expansion` is changed, let's say by a function (e.g. `expansionPanelState(v)`), the "internal value" is updated by `v-model` too (data flow in opposite direction).